### PR TITLE
Control: Cache user products

### PR DESCRIPTION
**Pull request by leoetlino**
Since req.user.product is populated on every route that works on a specific account, and WHMCS is pretty slow, let's cache the product info to make things faster.

---
This change is [Reviewable](https://reviewable.io/reviews/innovate-technologies/itframe/48)

### CODE REVIEW

File: src/http/control/00-security.js [hunk 1]
```diff
@@ -3,12 +3,32 @@ import wait from "wait.for";
...
 const InvalidatedTokens = requireFromRoot("components/control/invalidatedTokensDatabase.js");
 const controlUser = requireFromRoot("components/control/controlUser.js");
 const publicKey = fs.readFileSync(global.appRoot + "/keys/controlPublicKey.pem");
 const cert = fs.readFileSync(global.appRoot + "/keys/controlSigningKey.pem");
 
+async function invalidateCacheForEmail(email) {
+    await redisClient.del("user_products:" + email);
+}
+
+async function getProductsForEmail(email) {
```
Comment (meyskens) [inline src/http/control/00-security.js]: the right place to do this here and not in the middle class?

File: src/http/control/00-security.js [hunk 2]
```diff
@@ -146,6 +166,7 @@ export default function ({ app, expressJwt, jwt, wrap }) {
     app.post("/control/log-out", function (req, res) {
         wait.launchFiber(() => {
             wait.for(InvalidatedTokens.addToken, req.token);
+            invalidateCacheForEmail(req.user.email);
```
Comment (meyskens) [inline src/http/control/00-security.js]: maybe on login also? personally i never log out, only in

File: src/http/control/00-security.js (hunk 1 above)
Comment (leoetlino) [inline src/http/control/00-security.js]: Yeah, I'd prefer having this here as this is the only part that is performance sensitive. If we have this in the middle syntax, invalidation would get a bit more complicated (and there would need to be a way to bypass the cache when querying WHMCS).

File: src/http/control/00-security.js (hunk 2 above)
Comment (leoetlino) [inline src/http/control/00-security.js]: That sounds like a good idea. I'll change the invalidation to happen on login instead.

Comment (meyskens) [inline src/http/control/00-security.js]: okayio!

File: src/http/control/00-security.js (hunk 1 above)
Comment (meyskens) [inline src/http/control/00-security.js]: sure

**Pull request merged**